controllers/clusteroperator.go

```diff
@@ -144,40 +154,32 @@ func (r *ProvisioningReconciler) syncStatus(co *osconfigv1.ClusterOperator, cond
 	return err
 }
 
-// updateCOStatusDisabled updates the ClusterOperator's status to Disabled
-func (r *ProvisioningReconciler) updateCOStatusDisabled() error {
-	disabledMessage := "Operator is non functional"
-	availableMessage := "Operator is available while being disabled"
+func (r *ProvisioningReconciler) updateCOStatus(newReason StatusReason, msg, progressMsg string) error {
 
 	co, err := r.getOrCreateClusterOperator()
 	if err != nil {
 		r.Log.Error(err, "failed to get or create ClusterOperator")
 		return err
 	}
-
-	conds := []osconfigv1.ClusterOperatorStatusCondition{
-		setStatusCondition(osconfigv1.OperatorAvailable, osconfigv1.ConditionTrue, string(ReasonUnsupported), availableMessage),
-		setStatusCondition(OperatorDisabled, osconfigv1.ConditionTrue, string(ReasonUnsupported), disabledMessage),
-	}
-
-	return r.syncStatus(co, conds)
-}
-
-// updateCOStatusDegraded updates the ClusterOperator's Degraded
-// degradedReason should contain the current reason for the Operator to be marked in that state
-func (r *ProvisioningReconciler) updateCOStatusDegraded(degradedReason string, detailedError string) error {
-	degradedMessage := "Operator is Degraded"
-	progressingMessage := "Operator is Degraded while Progressing"
-
-	co, err := r.getOrCreateClusterOperator()
-	if err != nil {
-		return err
-	}
-
-	conds := []osconfigv1.ClusterOperatorStatusCondition{
-		setStatusCondition(osconfigv1.OperatorDegraded, osconfigv1.ConditionTrue, degradedReason, degradedMessage),
-		setStatusCondition(osconfigv1.OperatorProgressing, osconfigv1.ConditionTrue, detailedError, progressingMessage),
-		setStatusCondition(osconfigv1.OperatorAvailable, osconfigv1.ConditionFalse, "", ""),
+	conds := defaultStatusConditions()
```

Comment:
Starting from the defaults every time (rather than the current status) means that the "lastUpdated" timestamp will get updated every time, even when the condition value hasn't changed.

Reply:
don't think there is anything to worry about: https://github.com/openshift/cluster-baremetal-operator/blob/master/controllers/clusteroperator.go#L192
https://github.com/openshift/cluster-baremetal-operator/blob/master/vendor/github.com/openshift/library-go/pkg/config/clusteroperator/v1helpers/status.go#L18
so that copies only changes across, so i don't think there is a problem